slack_sdk/web/internal_utils.py

```diff
@@ -231,22 +231,29 @@ def _to_0_or_1_if_bool(v: Any) -> Union[Any, str]:
 
 
 def _warn_if_text_is_missing(endpoint: str, kwargs: Dict[str, Any]) -> None:
+    missing = "text"
     attachments = kwargs.get("attachments")
-    if attachments and isinstance(attachments, list):
+    # Note that this method does not verify attachments
+    # if the value is already serialized as a single str value.
+    if attachments is not None and isinstance(attachments, list):
+        # https://api.slack.com/reference/messaging/attachments
+        # Check if the fallback field exists for all the attachments
         if all(
             [
-                attachment.get("fallback")
-                and len(attachment.get("fallback").strip()) != 0
+                isinstance(attachment, dict)
```

Comment:
We can safely assume the element is a dict but just in case for avoiding any runtime errors